src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSEKSResourceDetector.cs

```diff
@@ -0,0 +1,148 @@
+
+                resourceAttributes = new List<KeyValuePair<string, object>>()
+                {
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudProvider, "aws"),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudPlatform, "aws_eks"),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeK8SClusterName, clusterName),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeContainerID, containerId),
+                };
+            }
+            catch (Exception ex)
+            {
+                AWSXRayEventSource.Log.ResourceAttributesExtractException(nameof(AWSEKSResourceDetector), ex);
+            }
+
+            return resourceAttributes;
+        }
+
+        private string GetEKSClusterName()
+        {
+            var credentials = this.GetEKSCredentials();
+
+            if (!this.IsEKSProcess(credentials))
+            {
+                return null;
+            }
+
+            var clusterInfo = this.GetEKSClusterInfo(credentials);
+            var clusterInfoObject = this.DeserializeResponse(clusterInfo);
+
+            return clusterInfoObject.Data?.ClusterName;
+        }
+
+        private bool IsEKSProcess(string credentials)
+        {
+            var httpClientHandler = this.CreateHttpClientHandler();
+            var awsAuth = ResourceDetectorUtils.SendOutRequest(AWSAuthUrl, "GET", new KeyValuePair<string, string>("Authorization", credentials), httpClientHandler).Result;
+            return string.IsNullOrEmpty(awsAuth);
+        }
+
+        private string GetEKSClusterInfo(string credentials)
+        {
+            var httpClientHandler = this.CreateHttpClientHandler();
+            return ResourceDetectorUtils.SendOutRequest(AWSClusterInfoUrl, "GET", new KeyValuePair<string, string>("Authorization", credentials), httpClientHandler).Result;
+        }
+
+        private HttpClientHandler CreateHttpClientHandler()
+        {
+            var httpClientHandler = new HttpClientHandler();
+            httpClientHandler.ClientCertificates.Add(new X509Certificate2(AWSEKSCertificatePath));
+            return httpClientHandler;
+        }
+
+        private AWSEKSClusterInformationModel DeserializeResponse(string response)
+        {
+            return ResourceDetectorUtils.DeserializeFromString<AWSEKSClusterInformationModel>(response);
+        }
+
+        private string GetEKSCredentials()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+
+            using (var streamReader = ResourceDetectorUtils.GetStreamReader(AWSEKSCredentialPath))
+            {
+                while (!streamReader.EndOfStream)
+                {
+                    stringBuilder.Append(streamReader.ReadLine().Trim());
+                }
+            }
+
+            return "Bearer " + stringBuilder.ToString();
+        }
+
+        private string GetEKSContainerId()
```

Comment:
You can unit test this with a fake Cgroup file and making the `MetadataFilePath` a parameter.

Example: https://github.com/aws/aws-xray-sdk-java/pull/141

Reply:
Added